File: src/web/Controllers/TransactionsController.cs
using System.IO;
using System.Threading.Tasks;
using core.fs;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using web.Utils;

namespace web.Controllers
{
    [ApiController]
    [Authorize]
    [Route("api/[controller]")]
    public class TransactionsController(ImportTransactions.Handler service) : ControllerBase
    {
        [HttpPost("import")]
        public async Task<ActionResult> Import(IFormFile file)
        {
            using var streamReader = new StreamReader(file.OpenReadStream());

            var content = await streamReader.ReadToEndAsync();

            var cmd = new ImportTransactions.Command(content, User.Identifier());

            return await this.OkOrError(service.Handle(cmd));
        }
    }
}

File: src/web.fs/Controllers/TransactionsController.fs
namespace web.Controllers

open System.IO
open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc
open Microsoft.AspNetCore.Authorization
open Microsoft.AspNetCore.Http
open core.fs
open web.Utils

[<ApiController>]
[<Authorize>]
[<Route("api/[controller]")>]
type TransactionsController(service: ImportTransactions.Handler) =
    inherit ControllerBase()

    [<HttpPost("import")>]
    member this.Import(file: IFormFile) = task {
        use streamReader = new StreamReader(file.OpenReadStream())
        let! content = streamReader.ReadToEndAsync()
        let cmd : ImportTransactions.Command = {Content = content; UserId = this.User.Identifier()}
        return! this.OkOrError(service.Handle(cmd))
    }
